L1Trigger/DemonstratorTools/plugins/GTTInputFileWriter.cc

```diff
@@ -0,0 +1,176 @@
+// -*- C++ -*-
+//
+// Package:    L1Trigger/DemonstratorTools
+// Class:      GTTInputFileWriter
+//
+/**\class GTTInputFileWriter GTTInputFileWriter.cc L1Trigger/DemonstratorTools/plugins/GTTInputFileWriter.cc
+
+ Description: Example EDAnalyzer class, illustrating how BoardDataWriter can be used to write I/O buffer files for hardware/firmware tests
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Thomas Williams <thomas.williams@stfc.ac.uk>
+//         Created:  Mon, 15 Feb 2021 00:39:44 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+#include "ap_int.h"
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTrack.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/Common/interface/View.h"
+
+#include "L1Trigger/DemonstratorTools/interface/BoardDataWriter.h"
+#include "L1Trigger/DemonstratorTools/interface/codecs/tracks.h"
+#include "L1Trigger/DemonstratorTools/interface/utilities.h"
+
+// If the analyzer does not use TFileService, please remove
+// the template argument to the base class so the class inherits
+// from  edm::one::EDAnalyzer<>
+// This will improve performance in multithreaded jobs.
+
+class GTTInputFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit GTTInputFileWriter(const edm::ParameterSet&);
+  ~GTTInputFileWriter() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> Track_t;
+
+  void beginJob() override;
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  edm::EDGetTokenT<edm::View<Track_t>> tracksToken_;
+  size_t eventCount_;
+  l1t::demo::BoardDataWriter fileWriter_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+// NOTE: At least some of the info from these constants will eventually come from config files
+constexpr size_t kGapLength(6);
```

Comment:
I know you expect to put at least some of these constants from config files. In the meantime, can these be put into system specific header files (i.e. GTT board config) so that they can be shared among the various emulators? I don't like the idea of each emulator having their own copy of these constants.